### PR TITLE
chore: centralize shared dev dependencies

### PR DESCRIPTION
**Pull request by andreapiras**
## 📎 Jira ticket
https://transferwise.atlassian.net/browse/DS-394

## ❓ Context  

Centralize dev dependencies

## 🚀 Changes 

Centralize less and jest dependencies
Centralize misc utilities

## ✅ Checklist

- [ ] All changes are covered by tests
- [ ] The changes are covered in docs
- [ ] The commits follow conventional commits standards

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -9,34 +9,39 @@
   "sideEffects": [
     "*.css"
   ],
+  "files": [
```
Comment (andreapiras) [inline packages/components/package.json:12]: whitelist files as opposite as npmignore which blacklist and it's harder to maintain. Files by default inlcudes also some. md and main https://docs.npmjs.com/files/package.json#files

File: packages/components/package.json [hunk 2]
```diff
@@ -9,34 +9,39 @@
...
     "build:clean": "rm -rf lib build",
-    "build:compile-css": "gulp lessCompiler",
+    "build:compile-css": "gulp lessCompiler --gulpfile='../../node_modules/@transferwise/gulp-config'",
     "build:umd": "NODE_ENV=umd rollup -c",
     "build:umd-nopolyfill": "NODE_ENV=umd-nopolyfill rollup -c",
     "build:es": "NODE_ENV=es babel src -d build/es/polyfill --ignore '**/*.spec.js','**/*.story.js'",
     "build:es-nopolyfill": "NODE_ENV=es-nopolyfill babel src -d build/es/no-polyfill --ignore '**/*.spec.js','**/*.story.js'",
     "build:copy-files": "cpx 'src/**/!(db)/*.{css,json,svg}' build/es/polyfill",
     "build:copy-files2": "cpx 'src/**/!(db)/*.{css,json,svg}' build/es/no-polyfill",
-    "test": "gulp lessCompiler && jest --env=jsdom --runInBand && yarn test:bundleSize",
```
Comment (andreapiras) [inline packages/components/package.json:33]: import of css are now mocked

File: .npmignore [hunk 3]
```diff
@@ -1,21 +0,0 @@
-node_modules
```
Comment (andreapiras) [inline .npmignore:1]: File removed in favour of package.json ```files``` property

File: packages/components/postcss.config.js [hunk 4]
```diff
@@ -1,18 +0,0 @@
-const postcssImport = require("postcss-import");
```
Comment (andreapiras) [inline packages/components/postcss.config.js:1]: Moved to root

File: packages/less-config/gulpfile.js [hunk 5]
```diff
@@ -9,19 +9,19 @@ var print = require('gulp-print').default;
 // Less Compiler
 const lessCompiler = () => {
   return gulp
-    .src(['src/**/*.less'])
+    .src([`${process.env.PWD}/src/**/*.less`])
```
Comment (andreapiras) [inline packages/less-config/gulpfile.js:12]: This resolved based on what directory is running the command

File: packages/css/package.json [hunk 6]
```diff
@@ -37,22 +37,7 @@
...
-    "gulp-less": "^4.0.1",
-    "gulp-plumber": "^1.2.1",
-    "gulp-postcss": "^8.0.0",
-    "gulp-print": "^5.0.2",
-    "gulp-rename": "^2.0.0",
-    "npm-run-all": "^4.1.5",
-    "postcss-custom-media": "^7.0.8",
-    "postcss-custom-properties": "^9.0.2",
-    "postcss-import": "^12.0.1",
+    "@transferwise/less-config": "^1.0.0",
```
Comment (helendurranttw) [inline packages/css/package.json:40]: 😌

File: packages/dynamic-flows/package.json [hunk 7]
```diff
@@ -45,32 +50,9 @@
...
-    "husky": "^4.2.3",
-    "jest": "24.8.0",
-    "jest-cli": "24.8.0",
-    "jest-fetch-mock": "^3.0.3",
-    "lint-staged": "^9.4.2",
-    "npm-run-all": "^4.1.5",
-    "postcss-custom-media": "^7.0.8",
-    "postcss-custom-properties": "^9.0.2",
-    "postcss-import": "^12.0.1"
+    "@transferwise/icons": "^1.0.0"
```
Comment (helendurranttw) [inline packages/dynamic-flows/package.json:55]: Are icons needed as a dev dep?

File: packages/dynamic-flows/package.json [hunk 8]
```diff
@@ -9,31 +9,36 @@
...
-    "dev:watch-less": "gulp lessWatcher",
+    "dev:watch-less": "gulp --gulpfile='../../node_modules/@transferwise/less-config' lessWatcher ",
     "dev:storybook": "start-storybook",
     "docs": "npm-run-all --parallel docs:*",
     "docs:babel": "NODE_ENV=es babel src -d build/es/polyfill --ignore '**/*.spec.js','**/*.story.js' -w",
     "docs:watch-less": "gulp lessWatcher",
     "build": "npm-run-all build:*",
     "build:clean": "rm -rf lib build",
-    "build:compile-css": "gulp lessCompiler",
+    "build:compile-css": "gulp lessCompiler --gulpfile='../../node_modules/@transferwise/less-config'",
```
Comment (helendurranttw) [inline packages/dynamic-flows/package.json:29]: I don't feel very comfortable with this but I don't have a better solution 🤔

Comment (helendurranttw) [reply packages/dynamic-flows/package.json:29]: Could you have a config file in this repo that points to the root, maybe? I guess that's not much better

Comment (helendurranttw) [reply packages/dynamic-flows/package.json:29]: Ahh! What about if you expose these as scripts from the packages? So test-config exposes a test script that will run the right thing for you? And you can just run it using yarn test or what have you

File: packages/test-config/README.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Things to add
```
Comment (helendurranttw) [inline packages/test-config/README.md:1]: What is this?

Comment (andreapiras) [reply packages/test-config/README.md:1]: Me adding all the things 😅

File: packages/dynamic-flows/package.json (hunk 8 above)
Comment (andreapiras) [reply packages/dynamic-flows/package.json:29]: Yeah I know is not great :( could not find any better

Comment (andreapiras) [reply packages/dynamic-flows/package.json:29]: Let me try what u suggested

File: packages/test-config/package.json [hunk 10]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@transferwise/test-config",
```
Comment (dozortsev) [inline packages/test-config/package.json:2]: do I understand correctly now this config will be published to NPM each time we make some changes there?

File: packages/test-config/README.md (hunk 9 above)
Comment (dozortsev) [reply packages/test-config/README.md:1]: it actually make sense to have small README.md file explaining what this tiny package for

for both test-config and less-config

File: packages/dynamic-flows/package.json (hunk 8 above)
Comment (dozortsev) [reply packages/dynamic-flows/package.json:29]: stupid question, have you tried just like this:

```
gulp lessCompiler --gulpfile='@transferwise/less-config'
```

Comment (andreapiras) [reply packages/dynamic-flows/package.json:29]: > stupid question, have you tried just like this:
> 
> ```
> gulp lessCompiler --gulpfile='@transferwise/less-config'
> ```
> 
> or
> 
> ```
> gulp lessCompiler --gulpfile=@transferwise/less-config/gulpfile.js
> ```

yup both didn't work

File: packages/css/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/css/package.json:40]: that file is slightly different. I don't know if it's worth it to dedicate more time to this ticket right now, Maybe we can create a separate ticket once this go out to take care of that

File: packages/less-config/package.json [hunk 11]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@transferwise/less-config",
+  "version": "1.0.0",
+  "description": "Neptune Web shared gulp css config files and dependencies",
+  "license": "Apache-2.0",
```
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/less-config/package.json:5]: yes that's the license we are using in all neptune-web

**Comment by andreapiras** [comment]
> @andreapiras does it make make also centralize babel configs? atm we have three of them, in [components](https://github.com/transferwise/neptune-web/blob/master/packages/components/babel.config.js) and [dynamic-flows](https://github.com/transferwise/neptune-web/blob/master/packages/dynamic-flows/babel.config.js) packages (very similar) and [`.babelrc`](https://github.com/transferwise/neptune-web/blob/master/packages/docs/.babelrc) in docs

Yes it makes sense. I don't have more time to dedicate to this ticket. So we can use jest and gulp to test out this configuration and then centralize babel

File: packages/test-config/package.json (hunk 10 above)
Comment (andreapiras) [reply packages/test-config/package.json:2]: correct

File: .gitignore [hunk 12]
```diff
@@ -32,7 +32,7 @@ lerna-debug.log
 
 # CSS
 dist
-docs-build
+*.css
```
Comment (andreapiras) [inline .gitignore:35]: Check if docs css works, eventually restrict to components and css

File: packages/components/package.json [hunk 13]
```diff
@@ -9,34 +9,39 @@
...
-    "test:watch": "jest --watch --env=jsdom",
+    "test": "jest --env=jsdom --runInBand --config='../../node_modules/@transferwise/test-config/jest.config.js' --rootDir && yarn test:bundleSize",
+    "test:watch": "jest --watch --env=jsdom --config='../../node_modules/@transferwise/test-config/jest.config.js' --rootDir",
     "test:bundleSize": "bundlesize",
     "lint": "eslint 'src/**/*.js'",
     "lint:fix": "npm run lint -- --fix"
```
Comment (andreapiras) [inline packages/components/package.json:40]: check this

File: packages/css/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/css/package.json:41]: Create one shared config

File: .gitignore (hunk 12 above)
Comment (andreapiras) [reply .gitignore:35]: Docs css are commitable
image

File: packages/less-config/README.md [hunk 14]
```diff
@@ -0,0 +1,4 @@
+# Neptune Web Configuration : Less
+
+@transferwise/test-config provides less configuration that is shared between packages.
```
Comment (helendurranttw) [inline packages/less-config/README.md:3]: Typo: `less-config`

File: packages/less-config/gulpfile.js [hunk 15]
```diff
@@ -9,19 +9,19 @@ var print = require('gulp-print').default;
...
+    .pipe(gulp.dest(`${process.env.PWD}/src`));
 };
 
 // Watch files
 const lessWatcher = () => {
-  gulp.watch(['src/**/*.less'], gulp.series(lessCompiler));
+  gulp.watch([`${process.env.PWD}/src/**/*.less`], gulp.series(lessCompiler));
 };
 
 exports.lessWatcher = lessWatcher;
```
[the remark this answers is not in the document]
Comment (helendurranttw) [reply packages/less-config/gulpfile.js:48]: @andreapiras

[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/less-config/gulpfile.js:48]: I gave a quick try and it seems too complicated for the benefits it offers. If we really need that please raise a ticket

**Pull request merged**